conda/gateways/connection/session.py

```diff
@@ -73,18 +77,37 @@ def get_channel_name_from_url(url: str) -> str | None:
     return Channel.from_url(url).canonical_name
 
 
+def get_plugin_headers(
+    url: str, http_headers_hooks: tuple[CondaHttpHeader, ...]
+) -> dict[str, str]:
+    """
+    Retrieves all HTTP headers set by plugins
+    """
+    url_parts = urlparse(url)
+
+    if url_parts.scheme == "https" or url_parts.scheme == "http":
+        return {
+            hook.header_name: hook.header_value
+            for hook in http_headers_hooks
+            if hook.header_hosts is None or url_parts.netloc in hook.header_hosts
+        }
+
+    return {}
```

Comment:
```suggestion
def get_request_headers(
    url: str, request_headers: tuple[CondaRequestHeader, ...]
) -> dict[str, str]:
    """
    Retrieves all HTTP headers set by plugins for the given URL
    """
    url_parts = urlparse(url)

    if url_parts.scheme in ("https", "http"):
        return {
            request_header.name: request_header.value
            for request_header in request_headers
            if request_header.hosts is None or url_parts.netloc in request_header.hosts
        }

    return {}
```

Reply:
One thing I'm concerned about is that URL parsing and validation is wrought with errors in my experience, can you see a failure state in which the hosts or the netloc could be empty? What do we do in that case?

Reply:
We could add an additional check to guard against that, but if the netloc is empty, don't we have bigger issues? This would simply not include any headers presuming that `request_header.hosts` is a either `None` or a set of host names, and then conda would encounter a different error later down the line.